usermanagerlib/src/main/java/org/pesmypetcare/usermanagerlib/clients/MedicationManagerClient.java

```diff
@@ -0,0 +1,216 @@
+        List<Medication> MedicationList = new ArrayList<>();
+        if (response.length() > 2) {
+            String jsonArray = response.substring(1, response.length() - 1);
+            String[] Medications = jsonArray.split(",\\{");
+            MedicationList.add(GSON.fromJson(Medications[0], Medication.class));
+            for (int i = 1; i < Medications.length; i++) {
+                Medications[i] = "{" + Medications[i];
+                MedicationList.add(GSON.fromJson(Medications[i], Medication.class));
+            }
+        }
+        return MedicationList;
+    }
+
+    /**
+     * Gets the data from all the Medications eaten by the pet between the initial and final date not including them.
+     * @param accessToken The personal access token for the account
+     * @param owner Username of the owner of the pets
+     * @param petName Name of the pet
+     * @param initialDate Initial Date
+     * @param finalDate Final Date
+     * @return The List containing all the Medications eaten by the pet in the specified time
+     */
+    public List<Medication> getAllMedicationsBetween(String accessToken, String owner, String petName,
+                                                     DateTime initialDate,
+                                         DateTime finalDate) throws ExecutionException, InterruptedException {
+        taskManager = taskManager.resetTaskManager();
+        taskManager.setTaskId(GET);
+        StringBuilder response =
+                taskManager.execute(BASE_URL + owner + dash + petName + "/between/" + initialDate
+                        + dash + finalDate, accessToken).get();
+        List<Medication> MedicationList = new ArrayList<>();
+        if (response.length() > 2) {
+            String jsonArray = response.substring(1, response.length() - 1);
+            String[] Medications = jsonArray.split(",\\{");
+            MedicationList.add(GSON.fromJson(Medications[0], Medication.class));
+            for (int i = 1; i < Medications.length; i++) {
+                Medications[i] = "{" + Medications[i];
+                MedicationList.add(GSON.fromJson(Medications[i], Medication.class));
+            }
+        }
+        return MedicationList;
+    }
+
+    /**
+     * Updates the Medication's field.
+     * @param accessToken The personal access token for the account
+     * @param owner Username of the owner of the pet
+     * @param petName Name of the pet
+     * @param date Date of the medication
+     * @param name Name of the medication
+     * @param field Name of the field to update
+     * @param value Value the field will have
+     * @return The response code
+     * @throws ExecutionException When the update fails
+     * @throws InterruptedException When the update is interrupted
+     */
+    public int updateMedicationField(String accessToken, String owner, String petName, DateTime date,
+                                     String name, String field,  Object value)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a single space to separate non-whitespace characters.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Libraries/pullRequest?prid=5403182)